### PR TITLE
feat: proper protos

### PR DESCRIPTION
**Pull request by grikomsn**
## Description

This PR overhauls generating custom protos to cosmjs types using telescope, then registering cctp amino and registry types. Also configures stargate account parser 

## Key Points

- [feat: vendorize submodules](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/a6643f5535aea0f2807063645de9245f7911bfc6)
- [feat: setup telescope](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/566ab97d9a384ae1672aef5a4334c9d72baf320e)
- [chore: add cctp amino reference](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/46c1cdf0418067e6e8a2deb0fcf0e9bc1f41ad49)
- [refactor: use kujira account parser](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/8c4f599e20381b149b6b0ed65b4211c3f3f72e0a)
- [fix: import codegen circle types](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/27708b202752743859da3ae49e6c1f9b9efddca2)
- [feat: allow external amino and registry types](https://github.com/skip-mev/skip-router-sdk/pull/95/commits/c68c27dd2e359ae1145cb1a7b4edd4d2d98ae7e7)

## Notes

- Tests are failing but working on next PR (#94)

### CODE REVIEW

File: scripts/codegen.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      path.resolve(cwd, "vendor/noble-cctp/proto"),
+    ],
+    outPath: path.resolve(cwd, "packages/core/src/codegen/"),
+    options: {
+      aminoEncoding: {
+        customTypes: {
+          useCosmosSDKDec: true,
+        },
+        enabled: true,
+        exceptions: {
```
Comment (bpiv400) [inline scripts/codegen.js:26]: How did we come up with the exceptions and the typeToAmino function? Are we imitating what we've seen work elsewhere or can you explain why this is required?

**Review by bpiv400 — APPROVED** [review]
Approved but asked one question about codegen.js

**Comment by bpiv400** [comment]
Probably shouldn't merge while this is breaking the e2e tests

File: scripts/codegen.js (hunk 1 above)
Comment (grikomsn) [reply scripts/codegen.js:26]: https://github.com/evmos/evmos/blob/v16.0.3/crypto/ethsecp256k1/ethsecp256k1.go#L31-L36

**Comment by grikomsn** [comment]
> Probably shouldn't merge while this is breaking the e2e tests

Fixed on #94

File: scripts/codegen.js (hunk 1 above)
Comment (grikomsn) [reply scripts/codegen.js:26]: Will add comments on codegen script, or even a dev docs explaining it.

**Pull request merged**